### PR TITLE
Enable capturing of pieces on game board

### PR DESCRIPTION
**Pull request by garyravago**
- Added invalid test case for move_to! method
- Modified move_to! method to return and not update file/rank position when of the same color
- Integrated move_to! method in Pieces controller to enable capturing of pieces
- Added view logic to not display captured pieces on game board in Game and Pieces views

_Showing captured pieces on sidebar has been added to Trello_

### CODE REVIEW

File: app/controllers/pieces_controller.rb [hunk 1]
```diff
@@ -5,16 +5,16 @@ def show
...
 
   def update
-    # puts params.inspect
     return render_not_found if current_piece.blank?
     if current_piece.user != current_user
       return render plain: 'Forbidden :(', status: :forbidden
     end
-    current_piece.update_attributes(piece_params)
+
+    current_piece.move_to!(params[:piece][:file],params[:piece][:rank])
```
Comment (hereismyusername) [inline app/controllers/pieces_controller.rb]: Add space after comma

File: spec/models/piece_spec.rb [hunk 2]
```diff
@@ -66,33 +66,57 @@
...
+    end
+
+    context "invalid case" do
+      before(:each) do
+        @game = FactoryGirl.build(:game)
+        @current_piece = FactoryGirl.create(:piece, file: 4, rank: 4, game: @game, color: 'white')
+        @piece_same_color = FactoryGirl.create(:piece, file: 5, rank: 5, game: @game, color: 'white')
+      end
+      
+      it "does not move the player's current piece when capturing a same color piece" do
```
Comment (hereismyusername) [inline spec/models/piece_spec.rb]: Consider something like it 'does not move to own-piece square' do

File: spec/models/piece_spec.rb [hunk 3]
```diff
@@ -66,33 +66,57 @@
...
+        @current_piece.move_to!(5, 5)
+
+        expect(@current_piece.file).to eq(4)
+        expect(@current_piece.rank).to eq(4)
+      end
+
+      it "does not capture the player's other same color piece" do
```
Comment (hereismyusername) [inline spec/models/piece_spec.rb]: Could be shortened to, e.g. it 'does not capture own piece' do

**Comment by hereismyusername** [comment]
- Added some minor comments
- Shortened test descriptions with reference to betterspecs.org

**Comment by garyravago** [comment]
@hereismyusername , thanks for the feedback.  I have committed the requested changes

**Pull request merged**